app/models/report.rb

```diff
@@ -478,8 +613,217 @@ def self.report_trending_search(report)
       report.data << {
         term: trend.term,
         unique_searches: trend.unique_searches,
-        ctr: (ctr * 100).ceil(1).to_s + "%"
+        ctr: (ctr * 100).ceil(1)
+      }
+    end
+  end
+
+  def self.report_moderators_activity(report)
+    report.labels = [
+      { type: :link, properties: [:username, :user_url], title: I18n.t("reports.moderators_activity.labels.moderator") },
+      { properties: [:flag_count], title: I18n.t("reports.moderators_activity.labels.flag_count") },
+      { type: :seconds, properties: [:time_read], title: I18n.t("reports.moderators_activity.labels.time_read") },
+      { properties: [:topic_count], title: I18n.t("reports.moderators_activity.labels.topic_count") },
+      { properties: [:post_count], title: I18n.t("reports.moderators_activity.labels.post_count") }
+    ]
+
+    report.modes = [:table]
+
+    report.data = []
+    mod_data = {}
+
+    User.real.where(moderator: true).pluck(:id, :username).each do |u|
+      mod_data[u[0]] = {
+        user_id: u[0],
+        username: u[1],
+        user_url: "/admin/users/#{u[0]}/#{u[1]}"
```

Comment:
Can we form this URL on the client side instead since it shares the same pattern. This way we just have to send the user_id and the username down which makes the payload smaller too. We'll also use less memory on the server side.

Reply:
I can ’t actually do this, as I want reports to be "standalone" and it would force us to carry more logic. Maybe I can reconsider this in the future.